### ISSUE TITLE
building with "cirrusci/windowsservercore:cmake" image ends with "'cmake' is not recognized as an internal or external command"

### ISSUE THREAD

**Issue by ped7g** [cirruslabs/docker-images-windows#13]
# Expected Behavior

cmake is found and executed

# Real Behavior

https://cirrus-ci.com/task/5133350539100160?command=build#L19
inside BAT file (not .yml script level, that one does `call some.bat`) I have commands:
```
C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build>mkdir build 
C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build>cd build 
C:\Users\ContainerAdministrator\AppData\Local\Temp\cirrus-ci-build\build>cmake --config Release .. 
'cmake' is not recognized as an internal or external command,operable program or batch file.
```

# Related Info

- [X] Task issue
  - OS: Windows
  - Task name: "windows Visual Studio 15 2017 build"
  - Script/cache name (if applies): "build_script"

Question/puzzling bit:
- this did work before, like 10 days back: https://cirrus-ci.com/task/4791252803649536?command=build#L19
- meanwhile I don't see any build-config related change in my commits (just source changes + I verified the build fails today even for the particular commit which did work before)
- I don't see any change on the image side, it's this one? https://hub.docker.com/layers/cirrusci/windowsservercore/cmake/images/sha256-08fa2c8ac2f7062664e09330737a917e5f6b8381d6075814304f6851c4c544ae?context=explore
- ^ that's "last pushed 2 months ago"

so what **did** change? The docker itself? And is it me using the image + batch files in wrong way, or why does it fail? I don't have any windows machine to tinker with it interactively, and I don't want to burn lot of community bandwidth/cpu-time by trying out things blindly, so if there's somebody actually understanding what am I doing and could point me in right direction, I will be grateful. :)

**Comment by fkorotkov** [comment]
Do you need mingw and msys2? Seems these packages are hanging the installation.

**Comment by ped7g** [comment]
I need GNU make, bash, GNU find, and diffutils (installed by chocolatey), I think the old working builds did find most of them inside MSYS2, so I didn't need to install them by chocolatey.

But I also need **cmake** to generate the VS build scripts, and the image has "cmake" in name, so while I could imagine why I'm missing stuff which was not written on the tin (bash, make, find..), I'm confused why cmake.exe does not work - feels like I'm overlooking some trivial thing or using it all in wrong way?

Also I can't understand what did change. My scripts are same, image is same... did some Cirrus-glue-part change meanwhile?

**Comment by ped7g** [comment]
thinking about my response... that's for the VS build task (needs: bash + make + find + cmake + MSCC).

I have other task testing mingw builds under windows, that one needs mingw32. (I have no idea what is happening in windows world, so if it's like mingw32 -> mingw64 move making the 32 version obsolete, that's no big deal for me, I will try to migrate).

Right now I'm using the same image for both tasks (as it was working before): `image: cirrusci/windowsservercore:cmake` + `os_version: 2019`

The batch files running the build are for VS task: `winbuild_vs15_2017.bat`
  (I deleted all empty/echo lines and lines doing debug output like `path`)
```
del Makefile
ren Makefile.win Makefile
mkdir build
cd build
cmake --config Release ..
dir /W
call "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Auxiliary\Build\vcvars64.bat"
msbuild sjasmplus.vcxproj /property:Configuration=Release
...
```
The `cmake` reports cmake is not found...

The mingw32 task has this batch file:
`winbuild_mingw32.bat`
```
choco install -y --no-progress diffutils
@call ContinuousIntegration\winbuild_set_msys2_path.bat
mingw32-make -f Makefile.win clean
mingw32-make -f Makefile.win -j3
...
```

The path batch `winbuild_set_msys2_path.bat`:
```
set PATH=C:\tools\msys64\usr\bin\;%PATH%;c:\tools\sjasmplus
bash --version
diff --version
cmp --version
find --version
gcc --version
mingw32-make --version
```
The bash is not there (or any other exe, neither mingw32-make, etc...)

Basically all my batch files are unable to find any ".exe" in those images.

Isn't it some kind of security change, do I need to use power script shells instead of DOS batch files?

(as I don't follow windows world, I may be doing something stupid, so please bear with me :) I haven't used windows for a decade+)

**Comment by ped7g** [comment]
I see there was some work done on windows images, I can see current "cmake" image having cmake.exe available.

MSYS2 is gone, but most of the stuff I did need it for is available in Git. There is mingw64 with gcc8.1 available too, having also `mingw32-make` for GNU make, which fully satisfy my needs for "mingw" builds.

(if you want GNU `bash/find/diff/cmp/...`, do `set PATH=C:\Program Files\Git\usr\bin\;%PATH%` early in the script, making things like `find.exe` from Git to take priority over Microsoft system `find.exe`).

So I'm going to close the issue, as the main issue reported about cmake missing is fixed. Sounds like some "but" is there... yes, there is. *But* I don't see MS C++ compiler in the image. Even if chocolatey package "visualstudio2019-workload-vctools" seems to be installed. I will open new issue for that (after checking first currently opened ones).
